### ISSUE TITLE
autodetection between qt 4 and qt 5 is pretty broken

### ISSUE THREAD

**Issue by jkarneges** [psi-plus/qconf#6]
I've seen two different kinds of errors when Qt 4 is not installed but Qt 5 is.

One where the configure script immediately fails while trying to locate qmake:
```
$ ./configure
Configuring Project ...
Verifying Qt build environment ... ./configure: 204: local: '/usr/lib/x86_64-linux-gnu/qt4/bin/qmake':: bad variable name
```

And another where conf compilation fails:
```
$ ./configure
Configuring Project ...
Verifying Qt build environment ... fail

Reason: There was an error compiling 'conf'.  See conf.log for details.

Be sure you have a proper Qt 4.0+ build environment set up.  This means not
just Qt, but also a C++ compiler, a make tool, and any other packages
necessary for compiling C++ programs.

If you are certain everything is installed, then it could be that Qt is not
being recognized or that a different version of Qt is being detected by
mistake (for example, this could happen if $QTDIR is pointing to a Qt 3
installation).  At least one of the following conditions must be satisfied:

 1) --qtdir is set to the location of Qt
 2) $QTDIR is set to the location of Qt
 3) QtCore is in the pkg-config database
 4) qmake is in the $PATH

This script will use the first one it finds to be true, checked in the above
order.  #3 and #4 are the recommended options.  #1 and #2 are mainly for
overriding the system configuration.

$ cat conf.log
/usr/lib/x86_64-linux-gnu/qt4/bin/moc -DHAVE_MODULES -DQT_NO_DEBUG -DQT_CORE_LIB -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4 -I. conf4.cpp -o conf4.moc
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DHAVE_MODULES -DQT_NO_DEBUG -DQT_CORE_LIB -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4 -I. -o conf4.o conf4.cpp
In file included from conf4.cpp:9:0:
conf4.h:12:18: fatal error: QtCore: No such file or directory
compilation terminated.
Makefile:211: recipe for target 'conf4.o' failed
make: *** [conf4.o] Error 1
```

Running with `--qtselect=5` works around both of these issues, but IMO this shouldn't be needed if only Qt 5 is installed.

**Comment by Ri0n** [comment]
Hi Justin!

I'm glad to see you near qconf/Psi dev again! :-)

I'm not sure how I can test it on my system. The only option for me is qtchooser. and I don't know how to make it believe Qt4 is not installed.

dizzynb ~ # qmake -v
QMake version 2.01a
Using Qt version 4.8.7 in /usr/lib64/qt4
dizzynb ~ # mv /usr/lib64/qt4 /usr/lib64/qt4.bak
dizzynb ~ # qmake -v
qmake: could not exec '/usr/lib64/qt4/bin/qmake': No such file or directory
dizzynb ~ # which qmake
/usr/bin/qmake
dizzynb ~ # ls -la /usr/bin/qmake
lrwxrwxrwx 1 root root 9 окт 31 21:28 /usr/bin/qmake -> qtchooser
dizzynb ~ # qmake -qt=5 -v
QMake version 3.0
Using Qt version 5.6.0 in /usr/lib64


Compiling qconf as a system package is not a problem since package maintainers usually set explicitly dependencies and configure arguments.

**Comment by jkarneges** [comment]
Hello there! Yes I recently ported Pushpin to Qt 5 so I had to upgrade qconf. :)

I'm not worried about compiling qconf itself, but this issue affects scripts generated by qconf too.

**Comment by Ri0n** [comment]
Ah I see. It seems I saw something like this before.

**Comment by jkarneges** [comment]
The first issue seems to be a bash-ism. "local" is not allowed in bourne shell. Second one I think is a local misconfiguration, so it can be ignored.

Main issue is we don't loop through available Qt options with qtchooser. Patch in just a moment.

**Comment by jkarneges** [comment]
Fixed in 0a084c1523fa4e744c0cfb91a4ea662027eb3cc0.
